projects/plugins/boost/app/modules/optimizations/critical-css/Critical_CSS.php

```diff
@@ -106,4 +107,16 @@ public function display_critical_css() {
 	public function update_total_problem_count( $count ) {
 		return ( new Critical_CSS_State() )->has_errors() ? ++$count : $count;
 	}
+
+	public function enqueue_scripts() {
+		$internal_path = apply_filters( 'jetpack_boost_asset_internal_path', 'app/assets/dist/' );
+
+		wp_enqueue_script(
+			'jetpack-boost-critical-css-gen',
+			plugins_url( $internal_path . 'critical-css-gen.js', JETPACK_BOOST_PATH ),
+			array(),
+			JETPACK_BOOST_VERSION,
+			true
+		);
```

Comment:
Instead of enqueuing the script, register the script and use is as a conditional dependency so that it's not included if the site is using premium (cloud css).

Reply:
By register you mean in the main Boost [enqueue_scripts](https://github.com/Automattic/jetpack/blob/d5ce31ae55515e6b55c82c8eb2e0550af1793dc7/projects/plugins/boost/app/admin/class-admin.php#L84) callback?

Also - this is inside the critical css module, so it's not active when using premium.

Reply:
I mean [`wp_register_script`](https://developer.wordpress.org/reference/functions/wp_register_script/) and add id as a dependency in the main jetpack-boost scripts

Reply:
Updated in fac7d17.